lib/semantics/resolve-names.cc

```diff
@@ -3767,35 +3784,33 @@ bool DeclarationVisitor::PassesLocalityChecks(
         "Assumed size array '%s' not allowed in a locality-spec"_err_en_US);
     return false;
   }
-  if (symbol.owner() == currScope()) {  // C1125 and C1126
-    SayAlreadyDeclared(name, symbol);
-    return false;
-  }
   // TODO: Check to see if the name can appear in a variable definition context
   return true;
 }
 
-Symbol *DeclarationVisitor::DeclareLocalEntity(const parser::Name &name) {
+Symbol *DeclarationVisitor::EnsureEnclosingEntity(const parser::Name &name) {
   Symbol *prev{FindSymbol(name)};
-  bool implicit{false};
   if (prev == nullptr) {
     // Declare the name as an object in the enclosing scope so that
     // the name can't be repurposed there later as something else.
     prev = &MakeSymbol(InclusiveScope(), name.source, Attrs{});
     ConvertToObjectEntity(*prev);
     ApplyImplicitRules(*prev);
-    implicit = true;
   }
+  return prev;
+}
```

Comment:
Please emit a warning when creating the shared variable.  It's going to be shared state in concurrent code with no initialization, so code after the loop that uses the shared variable probably will have no way to know whether it was ever written to or simply remains uninitialized.

Reply:
Thanks, @klausler .  Do you think that we should also warn for creating variable whose names appear in LOCAL and LOCAL_INIT locality-spec's?

Reply:
No, absolutely not.